### PR TITLE
fix(breadcrumb): account for menu direction in breadcrumb overflow caret

### LINKED ISSUE #9155: React BreadcrumbItem overflow position and shark fin incorrect
## What package(s) are you using?

- [ ] `carbon-components`
- [x] `carbon-components-react`

## Detailed description

> Describe in detail the issue you're having.
The overflow in a breadcrumb should have a sharkfin and an offset but does not because displayName is never set in this code 

image

This is probably because the recently introduced  `createComponentToggle` results in 

image

> Is this issue related to a specific component?

Breadcrumb using an overflow

> What did you expect to happen? What happened instead? What would you like to
> see changed?

As described above before regression.

> What browser are you working in?

Chrome.

> What version of the Carbon Design System are you using?

10.38.0 (latest)

> What offering/product do you work on? Any pressing ship or release dates we
> should be aware of?

@caronb/ibm-cloud-cognitive

### PR DESCRIPTION
**Pull request by emyarod**
Closes #9155

This PR fixes the ComponentToggle (ref #8863) display names to be the original component names (it was not being assigned before) to fix a regression in adding breadcrumb overflow menu caret styles. It also accounts for new overflow menu changes (support for more menu directions ref #7116)

#### Testing / Reviewing

Confirm the breadcrumb overflow menu caret appears as expected

### CODE REVIEW

**Comment by emyarod** [comment]
cc @tay1orjones, it looks like the displayName was supposed to be set as `FeatureToggle(${name})` but it doesn't seem to be possible to do this within `CreateComponentToggle` without dynamically setting the forwardRef function param. Currently no display name is set on the return value of `CreateComponentToggle` so for the time being I reset the component toggle display names to the original values. Let me know if we want to differentiate feature toggle display names from classic display names still

File: packages/react/src/internal/ComponentToggle.js [hunk 1]
```diff
@@ -9,22 +9,19 @@ import React from 'react';
...
       }
+      return null;
     }
 
     return React.createElement(classic, { ...props, ref: ref });
   }
 
-  ComponentToggle.displayName = `FeatureToggle(${name})`;
-
   return React.forwardRef(ComponentToggle);
```
Comment (tay1orjones) [inline packages/react/src/internal/ComponentToggle.js]: Would it work for the snapshot and the devtools if we applied the displayName to the result of the forwardRef call?

```js
// not sure if this one would still be needed
ComponentToggle.displayName = `FeatureToggle(${name})`;

const wrapped = React.forwardRef(ComponentToggle);
wrapped.displayName = `FeatureToggle(${name})`;
return wrapped;
```

or if the function name has to be dynamic, it looks like function names are configurable, but not writable. Would maybe need to be something like below?
```js
Object.defineProperty(ComponentToggle, 'name', {value: `FeatureToggle(${name})`, configurable: true});

return React.forwardRef(ComponentToggle);
```

**Review by tay1orjones — COMMENTED** [review]
Hm, this is confusing. React docs include [this](https://reactjs.org/docs/forwarding-refs.html#displaying-a-custom-name-in-devtools) as an explicit example, though the [issue from 2018](https://github.com/facebook/react/issues/13703#issuecomment-423346302) mentions that the function param must be named. 

The DataTable and TableToolbarMenu snapshots include the ForwardRef and FeatureToggle wrapper in the name, `ForwardRef(FeatureToggle(OverflowMenu)`, which I think is ultimately okay. I saw the same thing in the devtools in the initial implementation. 

But I get what you're saying - the public API snapshot doesn't actually show displayName as a property of the exported object.

File: packages/react/src/internal/ComponentToggle.js (hunk 1 above)
Comment (emyarod) [reply packages/react/src/internal/ComponentToggle.js]: yeah I think the first codeblock should work (and we can still remove the original `ComponentToggle.displayName` assignment). I'll test it out

**Review by tay1orjones — APPROVED** [review]
Sweet! 🎉  I'm glad that worked out

**Comment by emyarod** [comment]
bump @tw15egan @jnm2377 when you have a chance

**Review by tw15egan — APPROVED** [review]
LGTM 👍 ✅

**Pull request merged**